### PR TITLE
Updating Web project templates to ms.id.web 0.3.0-preview

### PR DESCRIPTION
**Pull request by jmprieur**
Summary of the changes
 - Updates AddMicrosoftIdentityWeb* names based on what 
   Identity and ASP.NET Core team agreed
 - Fixes issue with the MultiTenant case (tenantId = common)
- Uses the new reusable interfaces (AddMicrosoftGraph, and AddDownstreamApi)

This removes several files which are now part of Microsoft.Identity.Web
The projects will need to reference Microsoft.Identity.Web 0.3.0-preview

Addresses #bugnumber (in this specific format)

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 1]
```diff
@@ -73,38 +73,41 @@ public void ConfigureServices(IServiceCollection services)
...
-            services.AddDownstreamWebApiService(Configuration);
+                            .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
 #endif
 #if (GenerateGraph)
-            services.AddMicrosoftGraph(scopes, Configuration.GetValue<string>("CalledApi:CalledApiUrl"));
+                            .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
+#endif
+                            .AddInMemoryTokenCaches();
+#else
+                    ;
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:91]: Having a separate semicolon on a new line is weird in our templates. Let's move away from this pattern. What we'd want is something like:

```C#
            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
#if (GenerateApiOrGraph)
                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"))
                        .EnableTokenAcquisitionToCallDownstreamApi(initialScopes)
#if (GenerateApi)
                            .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
#endif
#if (GenerateGraph)
                            .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
#endif
                            .AddInMemoryTokenCaches();
#else
                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"));
#endif
```

This also applies to a bunch of other places in this PR.

FYI I'm going to be much more picky about styles in this PR since it's now RC1 instead of previews.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 2]
```diff
@@ -73,38 +73,41 @@ public void ConfigureServices(IServiceCollection services)
                 .AddEntityFrameworkStores<ApplicationDbContext>();
 #elif (OrganizationalAuth)
 #if (GenerateApiOrGraph)
-            string[] scopes = Configuration.GetValue<string>("CalledApi:CalledApiScopes")?.Split(' ');
+            string[] initialScopes = Configuration.GetValue<string>("DownstreamApi:Scopes")?.Split(' ');
+
 #endif
+            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"))
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:80]: Let's use the indentation pattern that's consistent with the rest of the file which is 4 spaces. Don't try to align with the previous line. As an example, see line 72-73 for where we do this. 

```suggestion
            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
                .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"))
```

Note this applies to many other places in this PR.

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs [hunk 3]
```diff
@@ -55,7 +55,17 @@ public async Task<IEnumerable<WeatherForecast>> Get()
         {
             HttpContext.VerifyUserHasAnyAcceptedScope(scopeRequiredByApi);
 
-            string downstreamApiResult = await _downstreamWebApi.CallWebApiAsync();
+            var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                string apiResult = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs:61]: Use `var` instead of string.Same with line 66.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 4]
```diff
@@ -73,38 +73,41 @@ public void ConfigureServices(IServiceCollection services)
                 .AddEntityFrameworkStores<ApplicationDbContext>();
 #elif (OrganizationalAuth)
 #if (GenerateApiOrGraph)
-            string[] scopes = Configuration.GetValue<string>("CalledApi:CalledApiScopes")?.Split(' ');
+            string[] initialScopes = Configuration.GetValue<string>("DownstreamApi:Scopes")?.Split(' ');
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:76]: Can we use `var` instead of `string[]`?

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs [hunk 5]
```diff
@@ -37,7 +37,7 @@ public class WeatherForecastController : ControllerBase
         private readonly ILogger<WeatherForecastController> _logger;
 
         // The Web API will only accept tokens 1) for users, and 2) having the access_as_user scope for this API
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs:39]:
```suggestion
        // The Web API will only accept tokens 1) for users, and 2) having the "api-scope" for this API
```

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/.template.config/dotnetcli.host.json [hunk 6]
```diff
@@ -32,6 +32,10 @@
       "longName": "tenant-id",
       "shortName": ""
     },
+    "DefaultScope": {
+      "longName": "default-scope",
+      "shortName": "s"
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/.template.config/dotnetcli.host.json:37]: I would avoid setting a shortname here, especially a short one such as `s`. Given that this isn't a required argument that's used in all scenarios, for example, templates where auth is not configured, we don't want to reserve a shortname.

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs [hunk 7]
```diff
@@ -9,15 +9,9 @@
 #endif
 using Microsoft.AspNetCore.Mvc;
 #if (OrganizationalAuth || IndividualB2CAuth)
+using Microsoft.AspNetCore.Authentication.JwtBearer;
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs:12]: We try to follow alphabetical order in the using statements. In this case let's swap line 12 and line 13. Also double check the imports of the other files.

**Review by JunTaoLuo — COMMENTED** [review]
Some minor style feedback.

**Comment by JunTaoLuo** [comment]
Some rough notes on how to test the new templates from a PR build. I'm omitting some small details but let me know if any of the steps are confusing.

0. Get the latest RC1 SDK (this step might be optional) and add it to the path, alternatively run all dotnet commands from that directory:
  https://dotnetcli.azureedge.net/dotnet/Sdk/5.0.100-rc.1.20379.10/dotnet-sdk-5.0.100-rc.1.20379.10-win-x64.zip
1. Download the templates nupkg
  The package can be found under the published artifacts of the azdo build under `Windows_Packages/Release/Shipping/Microsoft.DotNet.Web.ProjectTemplates.5.0.5.0.0-ci.nupkg`. For example: https://dev.azure.com/dnceng/public/_build/results?buildId=790308&view=artifacts&type=publishedArtifacts. This is an unrelated build for illustration purposes since this PR's build hasn't finished yet so don't actually use those templates.
2. Uninstall the existing project template
  `dotnet new -u Microsoft.DotNet.Web.ProjectTemplates.5.0`
3. Install new project template
  `dotnet new -i C:\Users\johluo\Downloads\Microsoft.DotNet.Web.ProjectTemplates.5.0.5.0.0-ci.nupkg`
4. Run your verification script/tool with the new templates

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs (hunk 7 above)
Comment (dougbu) [reply src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs:12]: nit: &hellip; but place System `using`s first

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs (hunk 1 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs]: Done.

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs (hunk 3 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Controllers/WeatherForecastController.cs]: Done

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs (hunk 4 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs]: Done

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs (hunk 5 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs]: Applied

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs (hunk 7 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs]: Done;

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 8]
```diff
@@ -73,38 +73,41 @@ public void ConfigureServices(IServiceCollection services)
...
-            services.AddDownstreamWebApiService(Configuration);
+                        .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
 #endif
 #if (GenerateGraph)
-            services.AddMicrosoftGraph(scopes, Configuration.GetValue<string>("CalledApi:CalledApiUrl"));
+                        .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
+#endif
+                        .AddInMemoryTokenCaches();
+#else
+                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"));
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:91]: Looks like there's an extra 4 spaces here.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 9]
```diff
@@ -73,38 +73,41 @@ public void ConfigureServices(IServiceCollection services)
...
+                        .AddInMemoryTokenCaches();
 #else
-            services.AddMicrosoftWebAppAuthentication(Configuration, "AzureAdB2C");
+                .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAdB2C"));
 #endif
 #endif
 #if (OrganizationalAuth || IndividualB2CAuth)
             services.AddControllersWithViews()
-                .AddMicrosoftIdentityUI();
+                    .AddMicrosoftIdentityUI();
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:110]: Indents

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 10]
```diff
@@ -116,7 +119,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddRazorPages();
 #if (OrganizationalAuth || IndividualB2CAuth)
             services.AddServerSideBlazor()
-                .AddMicrosoftIdentityConsentHandler();
+                        .AddMicrosoftIdentityConsentHandler();
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:122]: Indents

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Startup.cs [hunk 11]
```diff
@@ -63,31 +63,29 @@ public void ConfigureServices(IServiceCollection services)
...
+                        .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
 #endif
 #if (GenerateGraph)
-            services.AddMicrosoftGraph(Configuration.GetValue<string>("CalledApi:CalledApiScopes")?.Split(' '),
-                                       Configuration.GetValue<string>("CalledApi:CalledApiUrl"));
+                        .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
+#endif
+                        .AddInMemoryTokenCaches();
+#else
+                    .AddMicrosoftIdentityWebApi(Configuration.GetSection("AzureAd"));
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Startup.cs:78]: Extra indent

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Index.cshtml.cs [hunk 12]
```diff
@@ -36,7 +36,17 @@ public class IndexModel : PageModel
...
-            ViewData["ApiResult"] = await _downstreamWebApi.CallWebApiAsync();
+            var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                string apiResult = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+                ViewData["ApiResult"] = apiResult;
+            }
+            else
+            {
+                string error = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Index.cshtml.cs:47]: var

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Index.cshtml.cs [hunk 13]
```diff
@@ -36,7 +36,17 @@ public class IndexModel : PageModel
 
         public async Task OnGet()
         {
-            ViewData["ApiResult"] = await _downstreamWebApi.CallWebApiAsync();
+            var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                string apiResult = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Index.cshtml.cs:42]: var

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs [hunk 14]
```diff
@@ -73,28 +63,36 @@ public void ConfigureServices(IServiceCollection services)
...
+                        .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
 #endif
 #if (GenerateGraph)
-            services.AddMicrosoftGraph(Configuration.GetValue<string>("CalledApi:CalledApiScopes")?.Split(' '),
-                                       Configuration.GetValue<string>("CalledApi:CalledApiUrl"));
+                        .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
+#endif
+                        .AddInMemoryTokenCaches();
+#else
+                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"));
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs:81]: Extra indent

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs [hunk 15]
```diff
@@ -105,11 +103,11 @@ public void ConfigureServices(IServiceCollection services)
                 options.FallbackPolicy = options.DefaultPolicy;
             });
             services.AddRazorPages()
-                .AddMvcOptions(options => {})
-                .AddMicrosoftIdentityUI();
+                    .AddMvcOptions(options => {})
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs:106]: indents

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs [hunk 16]
```diff
@@ -105,11 +103,11 @@ public void ConfigureServices(IServiceCollection services)
...
+                    .AddMicrosoftIdentityUI();
 #elif (IndividualB2CAuth)
             services.AddRazorPages()
-                .AddMicrosoftIdentityUI();
+                    .AddMicrosoftIdentityUI();
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs:110]: indents

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Controllers/HomeController.cs [hunk 17]
```diff
@@ -38,11 +38,20 @@ public class HomeController : Controller
...
-        [AuthorizeForScopes(ScopeKeySection = "CalledApi:CalledApiScopes")]
+        [AuthorizeForScopes(ScopeKeySection = "DownstreamApi:Scopes")]
         public async Task<IActionResult> Index()
         {
-            ViewData["ApiResult"] = await _downstreamWebApi.CallWebApiAsync();
-
+            var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                string apiResult = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Controllers/HomeController.cs:47]: var

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Controllers/HomeController.cs [hunk 18]
```diff
@@ -38,11 +38,20 @@ public class HomeController : Controller
...
+                ViewData["ApiResult"] = apiResult;
+            }
+            else
+            {
+                string error = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Controllers/HomeController.cs:52]: var

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs [hunk 19]
```diff
@@ -4,20 +4,10 @@
...
-#if (MultiOrgAuth)
 using Microsoft.AspNetCore.Authentication.OpenIdConnect;
-#endif
-using Microsoft.AspNetCore.Authorization;
-#endif
-#if (IndividualB2CAuth)
 using Microsoft.Identity.Web;
 using Microsoft.Identity.Web.UI;
-using Microsoft.Identity.Web.TokenCacheProviders.InMemory;
+using Microsoft.AspNetCore.Authorization;
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:10]: sort

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs [hunk 20]
```diff
@@ -74,28 +64,37 @@ public void ConfigureServices(IServiceCollection services)
...
+                        .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
 #endif
 #if (GenerateGraph)
-            services.AddMicrosoftGraph(Configuration.GetValue<string>("CalledApi:CalledApiScopes")?.Split(' '),
-                                       Configuration.GetValue<string>("CalledApi:CalledApiUrl"));
+                        .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
+#endif
+                        .AddInMemoryTokenCaches();
+#else
+                    .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAd"));
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:82]: Extra indent

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs [hunk 21]
```diff
@@ -112,7 +111,7 @@ public void ConfigureServices(IServiceCollection services)
 #endif
 #if (OrganizationalAuth || IndividualB2CAuth)
            services.AddRazorPages()
-                .AddMicrosoftIdentityUI();
+                   .AddMicrosoftIdentityUI();
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:114]: Indent

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs [hunk 22]
```diff
@@ -45,32 +39,29 @@ public Startup(IConfiguration configuration)
...
+                        .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
 #endif
 #if (GenerateGraph)
-            services.AddMicrosoftGraph(Configuration.GetValue<string>("CalledApi:CalledApiScopes")?.Split(' '),
-                                       Configuration.GetValue<string>("CalledApi:CalledApiUrl"));
+                        .AddMicrosoftGraph(Configuration.GetSection("DownstreamApi"))
+#endif
+                        .AddInMemoryTokenCaches();
+#else
+                    .AddMicrosoftIdentityWebApi(Configuration.GetSection("AzureAd"));
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs:54]: Extra indent

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs [hunk 23]
```diff
@@ -45,32 +39,29 @@ public Startup(IConfiguration configuration)
...
-                .AddInMemoryTokenCaches();
-
-            services.AddDownstreamWebApiService(Configuration);
+                .AddMicrosoftIdentityWebApi(Configuration.GetSection("AzureAdB2C"))
+                    .EnableTokenAcquisitionToCallDownstreamApi()
+                        .AddDownstreamWebApi("DownstreamApi", Configuration.GetSection("DownstreamApi"))
+                        .AddInMemoryTokenCaches();
 #else
-            services.AddMicrosoftWebApiAuthentication(Configuration, "AzureAdB2C");
+                .AddMicrosoftIdentityWebApi(Configuration.GetSection("AzureAdB2C"));
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Startup.cs:64]: Extra indent

**Review by JunTaoLuo — APPROVED** [review]
Minor spacing fixes. I'm approving this now but @Tratcher @blowdart @javiercn @captainsafia should also take a look at the new API and template changes.

**Review by Tratcher — APPROVED** [review]
It looks in line with the agreed design.

File: src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs [hunk 24]
```diff
@@ -54,7 +54,17 @@ public async Task<IEnumerable<WeatherForecast>> Get()
         {
             HttpContext.VerifyUserHasAnyAcceptedScope(scopeRequiredByApi);
 
-            string downstreamApiResult = await _downstreamWebApi.CallWebApiAsync();
+            var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
```
Comment (captainsafia) [inline src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs:57]: Here and elsewhere we can structure the requests using:

```
using (var response = await _downstreamApi.CallWebApiForUserAsync...) {
   ....
}
```

Comment (Tratcher) [reply src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherForecastController.cs:57]: Or just
```
using var response = await _downstreamWebApi.CallWebApiForUserAsync("DownstreamApi").ConfigureAwait(false);
```

**Comment by JunTaoLuo** [comment]
I'll clean up the templates's indenting and react to feedback to get this merged today.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs (hunk 8 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:91]: Fixed

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Startup.cs (hunk 11 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Startup.cs:78]: Fixed

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs (hunk 19 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:10]: Fixed

**Comment by jmprieur** [comment]
I'm updating the PR right now (after testing)

**Comment by javiercn** [comment]
Hey folks, seems like the changes that we did originally broke the Blazor server individual auth template. Any chance we can get that addressed as part of this? https://github.com/dotnet/aspnetcore/issues/25353

**Comment by JunTaoLuo** [comment]
Sure, can you push a commit to this branch @javiercn

**Comment by jmprieur** [comment]
> Hey folks, seems like the changes that we did originally broke the Blazor server individual auth template. Any chance we can get that addressed as part of this? #25353

@javiercn: I was not aware of this issue. Let's fix it.

**Comment by JunTaoLuo** [comment]
Ah we don't have permission to push to your branch since it's in your fork. @javiercn I think there's no benefit to piggy-backing off of this PR. Care to send a separate one?

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 25]
```diff
@@ -73,38 +73,41 @@ public void ConfigureServices(IServiceCollection services)
...
+                        .AddInMemoryTokenCaches();
 #else
-            services.AddMicrosoftWebAppAuthentication(Configuration, "AzureAdB2C");
+                .AddMicrosoftIdentityWebApp(Configuration.GetSection("AzureAdB2C"));
 #endif
 #endif
 #if (OrganizationalAuth || IndividualB2CAuth)
             services.AddControllersWithViews()
-                .AddMicrosoftIdentityUI();
+                        .AddMicrosoftIdentityUI();
```
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:110]:
```suggestion
                .AddMicrosoftIdentityUI();
```

Sorry I mean that these extra indentations should be removed. As in revert the change to this line.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs (hunk 10 above)
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:122]: Ditto

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs (hunk 15 above)
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs:106]: Ditto, revert the extra indentations here and on lines 107 and 110

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs (hunk 19 above)
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:10]: Hmm seems like it isn't sorted, move this line to line 8

**Comment by javiercn** [comment]
@jmprieur thanks for taking care of it.

> Ah we don't have permission to push to your branch since it's in your fork. @javiercn I think there's no benefit to piggy-backing off of this PR. Care to send a separate one?

I'm fine having it as a separate PR as long as we fix it, hopefully before Tuesday to avoid going through full ask-mode.

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs (hunk 21 above)
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:113]: Ditto, revert indentation.

**Review by JunTaoLuo — APPROVED** [review]
I realized my previous comment was misleading.

**Comment by jmprieur** [comment]
> Ah we don't have permission to push to your branch since it's in your fork. @javiercn I think there's no benefit to piggy-backing off of this PR. Care to send a separate one?

@JunTaoLuo @javiercn : I just gave you permissions to my fork.
If anybody needs access, don't hesitate to ask (or give them. I'll make you admin when you have accepted the invite)

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs (hunk 25 above)
Comment (jmprieur) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:110]: Oh, I see :). Will do.
